integration-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java

```diff
@@ -214,13 +227,46 @@ protected ChannelImpl createChannel() {
           throw new RuntimeException(ex);
         }
         SslContext sslContext = null;
-        if (useTestCa) {
+        if (useTls) {
           try {
-            sslContext = SslContext.newClientContext(Util.loadCert("ca.pem"));
+            SslProvider provider = null;
+            File trustCertChainFile = null;
+            TrustManagerFactory trustManagerFactory = null;
+            File keyCertChainFile = null;
+            File keyFile = null;
+            String keyPassword = null;
+            KeyManagerFactory keyManagerFactory = null;
+            Iterable<String> ciphers = null;
+            CipherSuiteFilter cipherFilter = IdentityCipherSuiteFilter.INSTANCE;
+            ApplicationProtocolConfig apn = null;
+            long sessionCacheSize = 0;
+            long sessionTimeout = 0;
+
+            if (useTestCa) {
+              trustCertChainFile = Util.loadCert("ca.pem");
+            }
+
+            if (useTestClientCert) {
+              // OpenSSL provider does not support client keys??
+              provider = SslProvider.JDK;
+
+              keyCertChainFile = Util.loadCert("client.pem");
+              keyFile = Util.loadCert("client.key");
+              keyPassword = null;
+            }
+
+            ciphers = Util.reasonableCiphers();
```

Comment:
This doesn't work as well as it seems it should work. Namely, IdentityCipherSuiteFilter does _no_ filtering, even for unsupporting ciphers. Thus, you can end up requesting unsupported ciphers and the SSLEngine can hang forever. The proper fix seems to be to fix IdentityCipherSuiteFilter to filter unsupported ciphers.